packages/remix-dev/compiler.ts

```diff
@@ -356,6 +356,7 @@ async function createBrowserBuild(
     sourcemap: options.sourcemap,
     metafile: true,
     incremental: options.incremental,
+    mainFields: ["module", "browser", "main"],
```

Comment:
This is the main line that makes me a little nervous about this PR. If we're specifically building for the browser, and someone specifies the browser field in package.json, it feels like that should take precedence. Can you say a little more about why this change is necessary?

Reply:
My thought was that the "browser" field just means "this can run in the browser" and doesn't mean "ESM format". Since we are bundling to ESM, might as well take the module if it exists. On second thought though I've reverted it to be browser first. We need some definition for mainFields here though otherwise esbuild falls back to it's default behavior of "if something was `required` anywhere, always take "main". Without this we break client bundle tree shaking.